### PR TITLE
[MM-24337]: Fix the team and channel filtering UT to include empty team or channel

### PR DESCRIPTION
**Pull request by jials**
#### Summary
- Fix the team and channel filtering in search when filtering all the channels

#### Ticket Link
- Github issue: https://github.com/mattermost/mattermost-server/issues/14435
 - Fixes https://mattermost.atlassian.net/browse/MM-24337

### CODE REVIEW

**Comment by ethervoid** [comment]
@jials There is a test that is failing with the current changes `TestAutocompleteUsersInChannel/Check_OutOfChannel_results_with/without_VIEW_MEMBERS_permissions`. May you please take a look?

**Comment by jials** [comment]
Release note should not be required as this is an update on the unit tests.

File: store/searchtest/user_layer.go [hunk 1]
```diff
@@ -238,15 +238,15 @@ func testHonorChannelRestrictionsAutocompletingUsers(t *testing.T, th *SearchTes
 	})
 	t.Run("Autocomplete users with all channels restricted", func(t *testing.T) {
 		options := createDefaultOptions(true, false, false)
-		options.ViewRestrictions = &model.ViewUsersRestrictions{Channels: []string{}}
+		options.ViewRestrictions = &model.ViewUsersRestrictions{Teams: []string{}, Channels: []string{}}
```
Comment (ethervoid) [inline store/searchtest/user_layer.go:241]: You have to activate these tests for all the search engines doing `[]string{ENGINE_ALL}`

[here](https://github.com/mattermost/mattermost-server/pull/15912/files#diff-2c1c4b1eaee0b08c3dc2a4eeb4a0e01a8ff63024b1f701e82ea1519df4994559R23) and [here](https://github.com/mattermost/mattermost-server/pull/15912/files#diff-2c1c4b1eaee0b08c3dc2a4eeb4a0e01a8ff63024b1f701e82ea1519df4994559R28)

**Review by ethervoid — CHANGES_REQUESTED** [review]

File: store/searchtest/user_layer.go (hunk 1 above)
Comment (jials) [reply store/searchtest/user_layer.go:241]: The links are not working, but I assume that you are referring to the updated changes 😃

**Comment by ethervoid** [comment]
@jials We have a discrepancy here, not related to your changes but to our different search engines because in [this test](https://github.com/mattermost/mattermost-server/blob/v5.27.0/store/searchtest/user_layer.go#L282-L293):

- DB when filtering by one team and searching by another returns users that are in both teams
- Bleve and ES returns empty

We need to decide what case is the right for us but that should be another ticket with different changes. I'll make some changes and go back to ES and Bleve search engines so we can merge this PR of yours

// @mgdelacroix

**Review by ethervoid — APPROVED** [review]
LGTM! thank you for your contribution and your time @jials

**Comment by jasonblais** [comment]
@mgdelacroix kind reminder to help with reviews :)

**Review by mgdelacroix — APPROVED** [review]
LGTM! Thanks for your contribution @jials!! :rocket:

**Comment by srkgupta** [comment]
Hi @ethervoid or @mgdelacroix 
Can you please help with the QA Test Steps in the JIRA ticket: 
https://mattermost.atlassian.net/browse/MM-24337

I am currently not exactly sure what needs to be tested in this PR.

**Comment by ethervoid** [comment]
@srkgupta Given that this PR only changes the tests, initially we though we had to make other changes, if the test passes we can merge it

**Review by srkgupta — APPROVED** [review]
All unit tests passed. So approving the PR. Thanks for the confirmation @ethervoid

**Pull request merged**